R/utils.R

```diff
@@ -55,6 +55,21 @@ assert_text <- function(text) {
 #' @param pos The position of the token to start the search from.
 #' @importFrom rlang seq2
 next_non_comment <- function(pd, pos) {
+  if (length(pos) < 1 || is.na(pos) || pos >= nrow(pd)) return(integer(0))
   candidates <- seq2(pos + 1L, nrow(pd))
+  if (all(candidates %in% which(pd$token == "COMMENT"))) return(integer(0))
   setdiff(candidates, which(pd$token == "COMMENT"))[1]
 }
+
+#' Find the index of the last comment in the sequence of comments-only tokens
+#' after the token that has position `pos` in `pd`.
+#' @param pd A parse table.
+#' @param pos The position of the token to start the search from.
+extend_if_comment <- function(pd, pos) {
+  if (pos == nrow(pd)) return(pos)
+  if (pd[pos + 1,]$token == "COMMENT") {
```

Comment:
Probably not important here, but in general `pd$token[pos + 1]` requires a lot less work. Maybe we can search later for `,]` and `, ]` to find this and other instances?

Reply:
So that's for speed?
``` r
reprex::reprex_info()
#> Created by the reprex package v0.1.1.9000 on 2017-11-21

library("tidyverse")
#> Loading tidyverse: ggplot2
#> Loading tidyverse: tibble
#> Loading tidyverse: tidyr
#> Loading tidyverse: readr
#> Loading tidyverse: purrr
#> Loading tidyverse: dplyr
#> Conflicts with tidy packages ----------------------------------------------
#> filter(): dplyr, stats
#> lag():    dplyr, stats
tbl <- as_tibble(mtcars)
microbenchmark::microbenchmark(tbl$cyl[1], tbl[1, ]$cyl)
#> Unit: microseconds
#>          expr     min      lq      mean   median       uq       max neval
#>    tbl$cyl[1]  47.232  50.143  62.46642  60.7610  63.6915   190.264   100
#>  tbl[1, ]$cyl 254.001 298.076 569.41824 318.3655 338.8025 13161.350   100
```

Reply:
I just checked and I found `, ]$` and `,]$` only twice and I changed it as part of this PR to your suggested style.